packages/app-desktop/services/spellChecker/SpellCheckerServiceDriverNative.ts

```diff
@@ -37,23 +29,43 @@ export default class SpellCheckerServiceDriverNative extends SpellCheckerService
 		for (const toTry of languagesToTry) {
 			try {
 				this.session().setSpellCheckerLanguages([toTry]);
-				logger.info(`Set effective language from "${v}" to "${toTry}"`);
-				return;
+				return true;
 			} catch (error) {
 				logger.warn(`Failed to set language to "${toTry}". Will try the next one in this list: ${JSON.stringify(languagesToTry)}`);
 				logger.warn('Error was:', error);
 			}
 		}
 
 		logger.error(`Could not set language to: ${v}`);
```

Comment:
Git diff is a bit confusing. It was actually a part of the `setLanguage` function.
```js
public setLanguage(v: string) {
		// If we pass an empty array, it disables spell checking
		// https://github.com/electron/electron/issues/25228
		if (!v) {
			this.session().setSpellCheckerLanguages([]);
			return;
		}

		// The below function will throw an error if the provided language is
		// not supported, so we provide fallbacks.
		// https://github.com/laurent22/joplin/issues/4146
		const languagesToTry = [
			v,
			languageCodeOnly(v),
		].concat(localesFromLanguageCode(languageCodeOnly(v), this.availableLanguages));
		for (const toTry of languagesToTry) {
			try {
				this.session().setSpellCheckerLanguages([toTry]);
				logger.info(`Set effective language from "${v}" to "${toTry}"`);
				return;
			} catch (error) {
				logger.warn(`Failed to set language to "${toTry}". Will try the next one in this list: ${JSON.stringify(languagesToTry)}`);
				logger.warn('Error was:', error);
			}
		}
}
```
But since now we need to check not one multiple languages, if it will error out, I decided to split this to the separate function.

Reply:
I agree with splitting it into a separate function. I find it confusing that a method that checks if a language is supported logs messages about being unable to set the language, though!

(At some point someone might try to use isLanguageSupported in a context where the language isn't being set)

Reply:
After digging dipper into this function and the issue from which it's started, I found that we can get rid of it.

The problem was originally from issue https://github.com/laurent22/joplin/issues/4146. The cause of the issue is
that on macOS Electron function `getSpellCheckerLanguages` could return
language code, that is not supported by `setSpellCheckerLanguages`. Also
based on Electron documentation this two functions is no-op on macOS,
since it uses built-in spell checker, but they clearly not no-op, based
on the issues found on github like this one
https://github.com/electron/electron/issues/30215. During the initial
setup, if user first runs the app, and doesn't have
spellChecker.language set, we use `getSpellCheckerLanguages` to get
system language and set spell check to it. But since macOS could return
not supported code, the error will occur when trying to set it.

The workaround was to try to set the language, and if it trow a error,
try to set nearest supported language. This actually isn't needed, since
that we can check supported languages from
`availableSpellCheckerLanguages`, and it will always return languages,
that supported by `setSpellCheckerLanguages`. Since they are hardcoded
in the Chromium source code.
https://source.chromium.org/chromium/chromium/src/+/main:components/spellcheck/common/spellcheck_common.cc;l=30;drc=f1a93e5d28bef5c70d5552758d7aa6b85bc54418;bpv=0;bpt=1

So we can just check if the result of `getSpellCheckerLanguages` in the
list of `availableSpellCheckerLanguages`, and if not set it to our
default (en-US).